### ISSUE TITLE
Ordering extensions?

### ISSUE THREAD

**Issue by grofit** [showdownjs/showdown#170]
Hey I have an issue where I am trying to parse some content out and add some custom output around it, now it works fine, but for some reason I am noticing that underscores `_` are automatically getting converted to `<em>` when I do not want them to be, so is there a way for me to order my extension to run at the end or a way to tell it not to process this section any further?

**Comment by tivie** [comment]
Have a look at the Output Extensions. https://github.com/showdownjs/showdown/wiki/Extensions#creating-markdown-extensions

example:

```js
var demo = function(converter) {
  return [
    { type: 'output', regex: '/<(\/)?div/', replace: '<$1span' }
  ];
}
```

Output extensions are ran only after all core parsers and lang extensions are ran.

**Comment by grofit** [comment]
I did see them and they operate on the HTML rather than the original markdown content right? I was hoping I wouldn't need to go through the html and the markdown but will give it a go.

**Comment by tivie** [comment]
### Composite extensions

If you need to use a language extension, you can defer modifications for last using a combination of a language extension and a output extensions.

For instance, let's say you want to parse/capture every char sequence between `%start%` and `%end%`. However, that content should not be modified by showdown (all chars must be literal)

You can create a lang extension that looks for the pseudo tags `%start%` and `%end%`, capture everything in between and save it in a variable. 

Later, in the output extension, after showdown has finished parsing everything, you can put the content back unaltered.

Simple example (Fiddle here: http://jsfiddle.net/tivie/1rqr7xy8/)

```js
showdown.extension('myExt', function() {
    var matches = [];
    return [
        { 
            type: 'lang',
            regex: /%start%([^]+?)%end%/gi,
            replace: function(s, match) { 
                matches.push(match);
                var n = matches.length - 1;
                return '%PLACEHOLDER' + n + '%';
            }
        },
        {
            type: 'output',
            filter: function (text) {
                for (var i=0; i< matches.length; ++i) {
                    var pat = '<p>%PLACEHOLDER' + i + '% *<\/p>';
                    text = text.replace(new RegExp(pat, 'gi'), matches[i]);
                }
                //reset array
                matches = [];
                return text;
            }
        }
    ]
});
```

**Comment by grofit** [comment]
I assume these are edits as they were not there when I originally looked, this approach seems far easier as the `<em>` additions may just be one use case, thanks a lot for your help!

**Comment by tivie** [comment]
@grofit yeah. I added a fiddle and a code example. Easier to demonstrate.

**Comment by tivie** [comment]
Marking as closed since the issue seems to be resolved.

**Comment by grofit** [comment]
Yeah thanks, although I have noticed that some custom extensions I am using purely for allowing custom bindings within my own system getting butchered even with the above steps, I am not sure if it is the xss plugin though.

i.e the text leaves the output step looking like:

`<a class='some-class' data-bind='click: function(){ someMethod('some-string', 'other-string'); }'>Some Text</a>`

but then when it makes its way to the page if I view the element in firebug it looks like:

`<a }'="" 'some-string');="" other',="" string="" data-bind="click: function(){ someMethod(" class="some-class">Some Text</a>`

I assume once it leaves here the xss or some other filter hooks into it (only using tables, github, xss and this custom one) and destroys it even in the output phase. If it is the xss extension then maybe I will need to mod it to ignore my stuff.

**Comment by tivie** [comment]
It seems to be the xss filter. Just disable it and check if fixes the issue.

**Comment by grofit** [comment]
turns out its not the xss filter, I turned it off, completely removed it from the project, still happens, I am gonna try removing all other plugins see if it happens, It is unrelated to this but will possibly raise another issue.

**Comment by tivie** [comment]
The issue is browser related and not showdown related

check https://github.com/showdownjs/showdown/issues/171 for more info
